kratos/utilities/element_size_calculator.cpp

```diff
@@ -116,6 +116,105 @@ double ElementSizeCalculator<2,3>::MinimumElementSizeDerivative(
     KRATOS_CATCH("");
 }
 
+// Triangle2D6 version.
+template<>
+double ElementSizeCalculator<2,6>::MinimumElementSize(const Geometry<Node<3> >& rGeometry)
+{
+
+    /* Calculate node-edge distances */
+    const double x10 = rGeometry[1].X() - rGeometry[0].X();
+    const double y10 = rGeometry[1].Y() - rGeometry[0].Y();
+    const double x20 = rGeometry[2].X() - rGeometry[0].X();
+    const double y20 = rGeometry[2].Y() - rGeometry[0].Y();
+
+    // node 0, edge 12
+    double nx = -(y20-y10);
+    double ny = x20-x10;
+    double Hsq = x10*nx + y10*ny;
+    Hsq *= Hsq / (nx*nx + ny*ny);
+
+    // node 1, edge 20
+    nx = -y20;
+    ny = x20;
+    double hsq = x10*nx + y10*ny;
+    hsq *= hsq / (nx*nx + ny*ny);
+    Hsq = ( hsq < Hsq ) ? hsq : Hsq;
+
+    // node 2, edge 10
+    nx = -y10;
+    ny = x10;
+    hsq = x20*nx + y20*ny;
+    hsq *= hsq / (nx*nx + ny*ny);
+    Hsq = ( hsq < Hsq ) ? hsq : Hsq;
+
+    return std::sqrt(Hsq);
+}
+// Triangle2D6 version.
+template<>
+double ElementSizeCalculator<2,6>::MinimumElementSizeDerivative(
+    const unsigned int DerivativeNodeIndex,
+    const unsigned int DerivativeDirectionIndex,
+    const Geometry<Node<3> >& rGeometry)
+{
+    KRATOS_TRY
+
+    KRATOS_DEBUG_ERROR_IF(DerivativeNodeIndex > 5)
+        << "Invalid DerivativeNodeIndex [ DerivativeNodeIndex = " << DerivativeNodeIndex
+        << ", expected DerivativeNodeIndex < 6 ].\n";
+
+    KRATOS_DEBUG_ERROR_IF(DerivativeDirectionIndex > 1)
+        << "Invalid DerivativeDirectionIndex [ DerivativeDirectionIndex = " << DerivativeDirectionIndex
+        << ", expected DerivativeDirectionIndex < 2 ].\n";
+
+    /* Calculate node-edge distances */
+    const double x10 = rGeometry[1].X() - rGeometry[0].X();
+    const double x10_derivative = EdgeLengthDerivative(DerivativeNodeIndex, DerivativeDirectionIndex, 1, 0, 0);
+
+    const double y10 = rGeometry[1].Y() - rGeometry[0].Y();
+    const double y10_derivative = EdgeLengthDerivative(DerivativeNodeIndex, DerivativeDirectionIndex, 1, 0, 1);
+
+    const double x20 = rGeometry[2].X() - rGeometry[0].X();
+    const double x20_derivative = EdgeLengthDerivative(DerivativeNodeIndex, DerivativeDirectionIndex, 2, 0, 0);
+
+    const double y20 = rGeometry[2].Y() - rGeometry[0].Y();
+    const double y20_derivative = EdgeLengthDerivative(DerivativeNodeIndex, DerivativeDirectionIndex, 2, 0, 1);
+
+    // node 0, edge 12
+    double nx = -(y20-y10);
```

Comment:
const when possible

Reply:
These variables cannot be `const` since are modified below